app/policies/pair_request_policy.rb

```diff
@@ -0,0 +1,48 @@
+class PairRequestPolicy < ApplicationPolicy
+  alias_method :pair_request, :record
+
+  def index?
+    user_is_owner? || user.admin?
+  end
+
+  def show?
+    user_is_owner? || user.admin?
+  end
+
+  def create?
+    true
+  end
+
+  def accept?
+    pair_request.pending? && user == pair_request.invitee
+  end
+
+  def reject?
+    user_is_owner? && (pair_request.pending? || pair_request.accepted?)
+  end
```

Comment:
I feel like because we are going with different controllers for different actions to keep things crud, we should do the same with the policy. The class should be like `PairRequest::AcceptancePolicy`. Thoughts on this?

Reply:
I can probably find a workaround, but Pundit is a bit unfriendly to these more "abstract" CRUD operations. For example, in the view, it *needs* this to happen:
```rb
<% if policy(object).action? %>
  <%= render stuff %>
<% end %>
```
But an `Acceptance` isn't an actual domain model as we've written it. The only concrete object I can pass in is a `pair_request` instance, and getting `policy(pair_request)` to look at a different policy than `PairRequestPolicy` seems difficult and potentially impossible. There are "headless" policies, but then those don't use records at all.

But basically I need a policy that can use a pair request as the record, but isn't `PairRequestPolicy` -- and have that integrate with the controllers/views. I couldn't figure it out. If you have ideas for how to go about it, I'd be glad to hear them out.

Reply:
hmm, seems like we could namespace the policy. we'd still be using the pair_request object but atleast we'd have a different file for the acceptances_controller?
let me give it a try

Reply:
Yes to headless. When I was looking through I also thought this code they gave might work:
```
def create
  @publication = find_publication # assume this method returns any model that behaves like a publication
  # @publication.class => Post
  authorize @publication, policy_class: PublicationPolicy
  @publication.publish!
  redirect_to @publication
end
```
We just tell it the policy class (idk if this would work though).